### PR TITLE
Fix: Rate limiting fallback to retryAfterHeader

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Fix: Rate limiting fallback to retryAfterHeader

## :bulb: Motivation and Context
Coses https://github.com/getsentry/sentry-dart/issues/657

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

## :crystal_ball: Next steps
https://github.com/getsentry/sentry-dart/issues/659

### CODE REVIEW

File: dart/analysis_options.yaml [hunk 1]
```diff
@@ -13,8 +13,9 @@ analyzer:
     # allow self-reference to deprecated members (we do this because otherwise we have
     # to annotate every member in every test, assert, etc, when we deprecate something)
     deprecated_member_use_from_same_package: warning
+    unnecessary_import: ignore
```
Comment (denrase) [inline dart/analysis_options.yaml:16]: Had the same issue with the current flutter release, but i fixed those imports in all the files. Should we really disable this?

File: dart/test/protocol/rate_limiter_test.dart [hunk 2]
```diff
@@ -166,6 +166,25 @@ void main() {
...
+      () {
+    final rateLimiter = fixture.getSUT();
+    fixture.dateTimeToReturn = 0;
+    final eventItem = SentryEnvelopeItem.fromEvent(SentryEvent());
+    final envelope = SentryEnvelope(
+      SentryEnvelopeHeader.newEventId(),
+      [eventItem],
+    );
+
+    rateLimiter.updateRetryAfterLimits(null, '50', 429);
```
Comment (denrase) [inline dart/test/protocol/rate_limiter_test.dart:181]: The test has the same expectation as the previous one. Is there an easy way to test the case where we don't have a null result after filter?

**Review by denrase — APPROVED** [review]
LGTM, only some questions.

**Review by brustolin — APPROVED** [review]
LGTM

File: dart/analysis_options.yaml (hunk 1 above)
Comment (marandaneto) [reply dart/analysis_options.yaml:16]: yep, I already created https://github.com/getsentry/sentry-dart/issues/659
so I'd fix in a different PR

File: dart/test/protocol/rate_limiter_test.dart (hunk 2 above)
Comment (marandaneto) [reply dart/test/protocol/rate_limiter_test.dart:181]: indeed but it's passing a value now so it is still worth testing.

**Pull request merged**